pkg/snmp/traps/status/status.go

```diff
@@ -22,6 +23,19 @@ var (
 func init() {
 	trapsExpvars.Set("Packets", &trapsPackets)
 	trapsExpvars.Set("PacketsAuthErrors", &trapsPacketsAuthErrors)
+
+	expvarcollector.RegisterExpvarReport("snmpTrapsStats", func() (interface{}, error) {
+		status := make(map[string]interface{})
+
+		metricsJSON := []byte(trapsExpvars.String())
+		metrics := make(map[string]interface{})
+		json.Unmarshal(metricsJSON, &metrics) //nolint:errcheck
+		if dropped := getDroppedPackets(); dropped > 0 {
+			metrics["PacketsDropped"] = dropped
+		}
+		status["metrics"] = metrics
+		return status, nil
+	})
```

Comment:
We've got a PR under review (#19789) that ports the snmp traps server to the new component system; in a componentized world, does this still need to happen during init() or should it be part of component setup? (and if it's part of component setup, is there a way to unregister it during component teardown?)

I imagine that eventually when everything is a component these will be exposed via a [fx value group](https://pkg.go.dev/go.uber.org/fx#hdr-Value_Groups); we could easily make our current status component expose it this way if that would be helpful.

Reply:
Hey @dplepage-dd. Thanks for the feedback and linking the PR for the snmp traps server component.

I'm still getting familiar with the agent-shared component framework, but my intuition tells me either we move all code from the init() function to the Start lifecycle hooks or as you suggested, we use fx value groups. I'm not sure which one of those would be best. (I'm still unclear how we are using fx value groups) 

> is there a way to unregister it during component teardown?

If we were to adopt this pattern we should provide an unregister callback function 😄 Great point

Reply:
I would imagine that the `fx value groups` would be helpful patterns for the status component so the rest of the components can easily register the expVar into the status component. Please correct me if that is not intention with your suggestion